### ISSUE TITLE
Error in opencl doubletranspose self-test on Tesla P100

### ISSUE THREAD

**Issue by jonasbardino** [bh107/bohrium#327]
I just ran the self-test on our new dual NVidia Tesla P100 system and got the following error:
```
bardino@bart:~$ BH_STACK=opencl python /usr/share/bohrium/test/python/run.py /usr/share/bohrium/test/python/tests/test_*.py
Testing /usr/share/bohrium/test/python/tests/test_accumulate.py/accumulate_primitives/vector (2.57s) ✓
Testing /usr/share/bohrium/test/python/tests/test_accumulate.py/accumulate_sum/func (16.34s) ✓
Testing /usr/share/bohrium/test/python/tests/test_accumulate.py/accumulate_sum/method (10.71s) ✓
Testing /usr/share/bohrium/test/python/tests/test_accumulate.py/accumulate_views/accumulate (11.29s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/arange/arange (1.92s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/array/list (0.01s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/array/scalar (0.01s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/array_create/copy (1.05s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/array_create/ones (0.84s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/array_create/random (1.19s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/array_create/zeros (0.02s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_create.py/seq_of_scalars/list_of_scalars (0.01s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/concatenate/same_shape (10.60s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/diagonal/diagonal (3.28s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/diagonal_axis/diagonal (20.82s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/fill/function (0.07s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/fill/method (0.00s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/fill/view (0.00s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/flatten/flatten (3.66s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/flatten/flatten_forder (3.32s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/flatten/flatten_self (2.77s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/flatten/ravel (2.87s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/overlapping/add (0.26s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/overlapping/identity (0.08s) ✓
Testing /usr/share/bohrium/test/python/tests/test_array_manipulation.py/transpose/doubletransposeError code: -36
terminate called after throwing an instance of 'cl::Error'
  what():  clFinish
Aborted
```

The system runs Ubuntu 16.04.2 with default numpy, NVidia and OpenCL setup. The bohrium and bohrium-opencl are nightlies from the PPA. CPU self-tests complete without error there.

The same opencl self-test seems to run fine on the our older NVidia GTX680 Ubuntu 14.04.5 system. However, it then fails in another self-test. I'll report that issue separately.

**Comment by dionhaefner** [comment]
Which driver are you using? I am having a similar issue with a GTX 1050Ti.

**Comment by jonasbardino** [comment]
At the time of the bug report I was using the 375.39 driver from the native Ubuntu package (http://packages.ubuntu.com/xenial/nvidia-375) along with the CUDA 7.5 packages (http://packages.ubuntu.com/xenial/nvidia-cuda-toolkit). 
I needed CUDA 8 for other purposes, so I installed that in parallel from NVIDIA's repository and was forced onto a slightly updated 375.51 driver along with it. That one works with various other things, but completely breaks Bohrium, to the extent where even the first self-test fails critically when trying to detect the OpenCL devices.

**Comment by omegahm** [comment]
Yup, I am currently looking into both problems. One should be fixed with #330 (It's because OpenCL reports multiple platforms, but only one of those have actual devices)

**Comment by dionhaefner** [comment]
That seems to have fixed it for me (tests pass). I am getting a different error now, but I'll open a separate issue for that as soon as I find the time to look into it.

**Comment by omegahm** [comment]
@jonasbardino Since I haven't heard more regarding this, I believe that it's fixed?

**Comment by jonasbardino** [comment]
The NVidia driver was updated to 175.66 from the main Ubuntu repository in the meantime. 
My tests there with either and CUDA7/8 and the latest Bohrium PPA packages indicate that you solved the problem.
Now I only get the apparently harmless error:
{{{
Testing /usr/share/bohrium/test/python/tests/test_accumulate.py/accumulate_primitives/vectorError code: -1
 (2.46s) ✓
}}}
and the remaining tests all complete.
